config/mbed/CMakeLists.txt

```diff
@@ -273,3 +274,34 @@ target_include_directories(${APP_TARGET} PRIVATE
 target_compile_definitions(${APP_TARGET} PRIVATE
                            ${CHIP_DEFINES}                      
 )
+
+if("capsense" IN_LIST MBED_TARGET_LABELS)
+  target_compile_definitions(${APP_TARGET} PRIVATE CAPSENSE_ENABLED)
+   set(CAPSENSE_PATH ${CHIP_ROOT}/third_party/capsense/repo)
+
+    target_include_directories(${APP_TARGET} PRIVATE
+    ${CHIP_ROOT}/examples/platform/mbed/capsense
+    ${CAPSENSE_PATH}
+)
+
+  
+  target_sources(${APP_TARGET} PRIVATE
+   ${CHIP_ROOT}/examples/platform/mbed/capsense/cycfg_capsense.c
+   ${CHIP_ROOT}/examples/platform/mbed/capsense/capsense.cpp
+   ${CAPSENSE_PATH}/cy_capsense_control.c
+   ${CAPSENSE_PATH}/cy_capsense_tuner.c
+   ${CAPSENSE_PATH}/cy_capsense_centroid.c
+   ${CAPSENSE_PATH}/cy_capsense_csx_v2.c
+   ${CAPSENSE_PATH}/cy_capsense_csd_v2.c
+   ${CAPSENSE_PATH}/cy_capsense_filter.c
+   ${CAPSENSE_PATH}/cy_capsense_generator_v3.c
+   ${CAPSENSE_PATH}/cy_capsense_processing.c
+   ${CAPSENSE_PATH}/cy_capsense_selftest.c
+   ${CAPSENSE_PATH}/cy_capsense_sensing_v2.c
+   ${CAPSENSE_PATH}/cy_capsense_sensing_v3.c
+   ${CAPSENSE_PATH}/cy_capsense_structure.c
+)
```

Comment:
Can you create a library for it. We can put it in https://github.com/ARMmbed/mbed-os-cypress-capsense-button .